File: api/app/models/product.rb
class Product < ApplicationRecord
    has_many :reviews
    has_many :cart_products
    has_many :orders

    belongs_to :brand
    belongs_to :category

    validates :title, :description, :price, :sku,  presence: true
    validates :title, :description, length: { minimum: 2 }
    validates :price, :average_rating, numericality: true
    validates :sku, uniqueness: true

    def update_average_rating
        average=0
        unless self.reviews.empty?
            total = self.reviews.sum(&:rating)
            average = (total / self.reviews.length).round
        end

        self.update(average_rating: average)
    end

end

File: src/components/Product.js
import React from 'react';
import { Image } from 'cloudinary-react';
import { CLOUD_NAME } from '../App';

const Product = props => {
    const { title, description, price, image_urls, category, sku } = props.product
    return (
        <>
            <h1>{title}</h1>
            <p>{price}</p>
    <p>{sku}</p>
            <Image cloudName={CLOUD_NAME} publicId={`${category}/${sku}/${image_urls[0]}`} />
            <p>{description}</p>
        </>
    )
}
export default Product;